Azure.ResourceManager.Core/Resources/Identity.cs

```diff
@@ -171,6 +196,11 @@ public static void Serialize(Utf8JsonWriter writer, Identity identity)
             writer.Flush();
         }
 
+        /// <summary>
+        /// Detects if this Identity is equals to antoher Identity instance.
+        /// </summary>
+        /// <param name="other"> Identity object to compare. </param>
+        /// <returns> True or False depending if they are equals. </returns>
```

Comment:
True if they are equal, otherwise False. (use the same as what it is in SystemAssignedIdentity.cs)